mammoth/opts.py

```diff
@@ -681,9 +500,25 @@ def _add_train_general_opts(parser):
     group.add(
         '--keep_checkpoint', '-keep_checkpoint', type=int, default=-1, help="Keep X checkpoints (negative: keep all)"
     )
+    group.add('--train_steps', '-train_steps', type=int, default=100000, help='Number of training steps')
+    group.add(
+        '--single_pass', '-single_pass', action='store_true', help="Make a single pass over the training dataset."
```

Comment:
Single pass almost certainly doesn't work (maybe just silently ignored).

Implementing it in a multi-task setting would be nontrivial: tasks would need to be able to remove themselves from the task sampling when they run out of data.